contracts/LosslessControllerV3.sol

```diff
@@ -330,32 +333,35 @@ contract LosslessControllerV3 is Initializable, ContextUpgradeable, PausableUpgr
     // @notice Sets protection for an address with the choosen strategy.
     // @dev Strategies are verified in the guardian contract.
     // @dev This call is initiated from a strategy, but guardian proxies it.
-    function setProtectedAddress(address token, address protectedAddresss, ProtectionStrategy strategy) external onlyGuardian whenNotPaused {
-        Protection storage protection = tokenProtections[token].protections[protectedAddresss];
+    function setProtectedAddress(ILERC20 token, address protectedAddress, ProtectionStrategy strategy) override external onlyGuardian whenNotPaused {
+        Protection storage protection = tokenProtections[token].protections[protectedAddress];
         protection.isProtected = true;
         protection.strategy = strategy;
-        emit ProtectedAddressSet(token, protectedAddresss, address(strategy));
+        emit NewProtectedAddress(address(token), protectedAddress, address(strategy));
     }
 
     // @notice Remove the protection from the address.
     // @dev Strategies are verified in the guardian contract.
     // @dev This call is initiated from a strategy, but guardian proxies it.
-    function removeProtectedAddress(address token, address protectedAddresss) external onlyGuardian whenNotPaused {
-        delete tokenProtections[token].protections[protectedAddresss];
-        emit RemovedProtectedAddress(token, protectedAddresss);
+    function removeProtectedAddress(ILERC20 token, address protectedAddress) override external onlyGuardian whenNotPaused {
+        require(isAddressProtected(token, protectedAddress), "LSS: Address not protected");
+        delete tokenProtections[token].protections[protectedAddress];
+        emit RemovedProtectedAddress(address(token), protectedAddress);
     }
 
     /// @notice This function will return the non-settled tokens amount
-    /// @param token Address corresponding to the token being held
+    /// @param _token Address corresponding to the token being held
     /// @param account Address to get the available amount
     /// @return Returns the amount of locked funds
-    function getLockedAmount(address token, address account) public view returns (uint256) {
-        uint256 lockedAmount;
+    function getLockedAmount(ILERC20 _token, address account) override public view returns (uint256) {
+        uint256 lockedAmount = 0;
         
-        LocksQueue storage queue = tokenScopedLockedFunds[token].queue[account];
+        LocksQueue storage queue = tokenScopedLockedFunds[_token].queue[account];
 
         uint i = queue.first;
-        while (i <= queue.last) {
+        uint256 lastItem = queue.last;
+
+        while (i <= lastItem) {
             ReceiveCheckpoint memory checkpoint = queue.lockedFunds[i];
             if (checkpoint.timestamp > block.timestamp) {
                 lockedAmount = lockedAmount + checkpoint.amount;
```

Comment:
Please add a comment for CVF-111. "Overflow is not possible because the account cannot receive more than the total supply of the token. Token supply cannot be larger than type(uint256).max."

Reply:
Addressed on commit d492c9a